### PR TITLE
docs(v2): Move sidebar content to its own page

### PR DESCRIPTION
**Pull request by ArtFlag**
## Motivation

Related to #3893 and #3894.

TOCs or sidebars are extremely important in any docs website.

In the current version, the sidebar info is inside the `intro` page and the sidebar sections are from H3 and deeper, meaning they do not appear in the right-hand TOC making it difficult to browse and understand. 

With this PR, the sidebar info is in its own page. It appears in the sidebar, and the on-page TOC is useful again. 

I've touched up the content a little bit to reduce word count, also see the `Test plan` section below.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Please take a solid look at the `Understanding sidebar items` section.
Task-oriented docs are generally much better for new users, so I've reworked the titles and description to tell us what each type does immediately, instead of telling us the name of each type first (also helps with SEO). 

Let me know if this is correct or if needs improvements, I'll take care of it swiflty.

The rest has only changed a little.

### CODE REVIEW

File: website/docs/sidebar.md [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+    'Category B': ['doc3'],
+  },
+};
+```
+
+By default, the doc page the user is reading will display the sidebar that it is part of. This can be customized with the [sidebar type](#understanding-sidebar-items).
+
+For example, with the above example, when displaying the `doc2` page, the sidebar will contain the items of `secondSidebar` only. Another example of multiple sidebars is the `API` and `Docs` sections on the Docusaurus website.
+
+For more information about sidebars and how they relate to doc pages, see [Navbar doc link](docs/api/themes/configuration/#navbar-doc-link).
```
Comment (slorber) [inline website/docs/sidebar.md:104]:
```suggestion
For more information about sidebars and how they relate to doc pages, see [Navbar doc link](./api/themes/theme-configuration.md#navbar-doc-link).
```

Netlify CI detects a broken link here.

image

For linking from one doc to another inside a given version, it's better to use relative paths to the md file.

**Comment by slorber** [comment]
Thanks, that looks like a good idea to me, but will be easier to review if the deploy preview works :)

**Comment by slorber** [comment]
@ArtFlag there is now a problem in the deploy preview:

https://deploy-preview-3899--docusaurus-2.netlify.app/classic/docs/sidebar

image


Also, can you comment in this PR the parts of the doc you modified?

As the content is both moved/updated, highlighting the major doc changes would be helpful, but overall it looks good

**Comment by ArtFlag** [comment]
Thanks for fixing the issues! 

I've mostly modified the section about the `type` property. 

The rest of the page has seen some modifications but mostly to reduce word count.

For example, for the shorthand note that you mentioned: we used to give the shorthand first and then explain what the "normal" version is.
I flipped it around because we talk about the side item object in this page, so didactically, we should probably show the non-shorthand version first... which actually shows the sidebar item object 😉 

No content has been deleted.

**Comment by slorber** [comment]
Thanks

Can you fix the git conflict and the admonition mistake so that I can merge this?

**Comment by ArtFlag** [comment]
It should be fine now. 

FYI I couldn't push my docs update because pre-commit thought it was an empty commit. Not sure if this update came from my rebase on master, but if there's a way to fix (ignore docs?) that would be nice 👍

**Comment by slorber** [comment]
thanks

not sure what you are talking about but that looks fine to merge :)

**Pull request merged**